src/blas/impl/KokkosBlas1_iamax_impl.hpp

```diff
@@ -0,0 +1,231 @@
+                   "It must be nonconst, because it is an output argument "
+                   "(we have to be able to write to its entries).");
+    static_assert (RV::rank == 0 && XV::rank == 1,
+                   "KokkosBlas::Impl::V_Iamax_Functor: "
+                   "RV must have rank 0 and XV must have rank 1.");
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()( const size_type & i, maxloc_type & lmaxloc ) const {
+	mag_type val = IPT::norm (m_x(i));
+    if( val > lmaxloc.val ) { lmaxloc.val = val; lmaxloc.loc = i; }
+  }
+};
+
+/// \brief Column-wise Iamax functor for multivectors.
+///
+/// \tparam RV 1-D output View
+/// \tparam XMV 2-D input View
+/// \tparam MagType Magnitude type
+/// \tparam SizeType Index type.  Use int (32 bits) if possible.
+template<class RV, class XMV, class MagType, class SizeType = typename XMV::size_type>
+struct MV_Iamax_FunctorVector
+{
+  typedef typename XMV::execution_space                         execution_space;
+  typedef SizeType                                              size_type;
+  typedef MagType                                               mag_type;
+  typedef typename XMV::non_const_value_type                    xvalue_type;
+  typedef Kokkos::Details::InnerProductSpaceTraits<xvalue_type> IPT;
+  typedef Kokkos::MaxLoc<mag_type,size_type>                         maxloc_reducer;
+  typedef typename Kokkos::MaxLoc<mag_type,size_type>::value_type    maxloc_type;
+  typedef typename Kokkos::TeamPolicy<execution_space>::member_type  member_type;
+
+  typename XMV::const_type m_x;
+  typename RV::non_const_type m_r;
+  
+  MV_Iamax_FunctorVector (const RV& r, const XMV& x) :
+    m_r (r), m_x (x)
+  {
+    static_assert (Kokkos::Impl::is_view<RV>::value,
+                   "KokkosBlas::Impl::MV_Iamax_FunctorVector: "
+                   "R is not a Kokkos::View.");
+    static_assert (Kokkos::Impl::is_view<XMV>::value,
+                   "KokkosBlas::Impl::MV_Iamax_FunctorVector: "
+                   "X is not a Kokkos::View.");
+    static_assert (Kokkos::Impl::is_same<typename RV::value_type,
+                   typename RV::non_const_value_type>::value,
+                   "KokkosBlas::Impl::MV_Iamax_FunctorVector: "
+                   "R is const.  It must be nonconst, because it is an output "
+                   "argument (we must be able to write to its entries).");
+    static_assert (RV::rank == 1 && XMV::rank == 2,
+                   "KokkosBlas::Impl::MV_Iamax_FunctorVector: "
+                   "RV must have rank 1 and XMV must have rank 2.");
+  }
+
+    KOKKOS_INLINE_FUNCTION
+    void operator() (const member_type &teamMember) const
+    {
+      const int lid = teamMember.league_rank();// teamId
+      const int tid = teamMember.team_rank(); // threadId
+
+      maxloc_type col_maxloc;
+      Kokkos::parallel_reduce( Kokkos::TeamThreadRange(teamMember, m_x.extent(0)), [&] (const int& i, maxloc_type& thread_lmaxloc) {
```

Comment:
We don't need to pass `int` by reference.  `int i` is fine.

Reply:
@mhoemmen It is fixed.